### PR TITLE
Map cleanup

### PR DESCRIPTION
**Pull request by pflopez**
- using untilDestroy for cleanup
- simplify some observables
- cleanup of async methods on templates
- moving some config to constants

### CODE REVIEW

File: src/interface/src/app/map/map.constants.ts [hunk 1]
```diff
@@ -26,3 +28,9 @@ export const LEGEND = {
     '#508295',
   ],
 };
+
+export const ERROR_SNACK_CONFIG: MatSnackBarConfig<any> = {
+  duration: 10000,
+  panelClass: ['snackbar-error'],
+  verticalPosition: 'top',
+};
```
Comment (lastminutediorama) [inline src/interface/src/app/map/map.constants.ts:36]: this will be useful, thanks

File: src/interface/src/app/map/map.component.ts [hunk 2]
```diff
@@ -345,12 +325,14 @@ export class MapComponent implements AfterViewInit, OnDestroy, OnInit, DoCheck {
...
 
     // Renders the selected region on the map.
-    this.selectedRegion$.subscribe((selectedRegion: Region | null) => {
-      var centerCoords = regionMapCenters(selectedRegion!);
-      map.instance?.setView(new L.LatLng(centerCoords[0], centerCoords[1]));
-      // Region highlighting disabled for now
-      // this.displayRegionBoundary(map, selectedRegion);
-    });
+    this.selectedRegion$
+      .pipe(untilDestroyed(this))
```
Comment (lastminutediorama) [inline src/interface/src/app/map/map.component.ts:329]: TIL... :tada:

File: src/interface/src/app/map/map.component.ts [hunk 3]
```diff
@@ -57,51 +46,72 @@ import { PlanCreateDialogComponent } from './plan-create-dialog/plan-create-dial
...
+  boundaryConfig$ = this.mapService.boundaryConfig$
+    .asObservable()
+    .pipe(untilDestroyed(this));
+  conditionsConfig$ = this.mapService.conditionsConfig$.asObservable();
+  selectedRegion$ = this.sessionService.region$.asObservable();
+
+  selectedMap$ = this.mapViewOptions$.pipe(
+    map((options) => this.maps[options.selectedMapIndex])
+  );
+  getOpacityForSelectedMap$ = this.selectedMap$.pipe(
```
Comment (lastminutediorama) [inline src/interface/src/app/map/map.component.ts:102]: nit:  If this is now a property, should we rename this to `opacityForSelectedMap` or even 'selectedMapOpacity`?

**Review by lastminutediorama — APPROVED** [review]
one nit/question, but this seems like a helpful tidying, thank you

File: src/interface/src/app/map/map.component.ts (hunk 3 above)
Comment (pflopez) [reply src/interface/src/app/map/map.component.ts:102]: yes! good catch

**Pull request merged**